docs/tools/purge-cache-for-changed-files.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+import subprocess
+import requests
+import os
+import time
+
+import warnings
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+warnings.simplefilter('ignore', InsecureRequestWarning)
+
+FNAME_START = '+++'
+
+CLOUDFLARE_URL = 'https://api.cloudflare.com/client/v4/zones/4fc6fb1d46e87851605aa7fa69ca6fe0/purge_cache'
+
+# we have changes in revision and commit sha on all pages
+# so such changes have to be ignored
+MIN_CHANGED_WORDS = 4
+
+
+def collect_changed_files():
+    proc = subprocess.Popen("git diff HEAD~1 --word-diff=porcelain | grep -e '^+[^+]\|^\-[^\-]\|^\+\+\+'", stdout=subprocess.PIPE, shell=True)
+    changed_files = []
+    current_file_name = ''
+    changed_words = []
+    while True:
+        line = proc.stdout.readline().decode('utf-8').strip()
+        if not line:
+            break
+        if FNAME_START in line:
+            if changed_words:
+                if len(changed_words) > MIN_CHANGED_WORDS:
+                    changed_files.append(current_file_name)
+                changed_words = []
+            current_file_name = line[6:]
+        else:
+            changed_words.append(line)
+    return changed_files
+
+
+def filter_and_transform_changed_files(changed_files):
+    result = []
+    for f in changed_files:
+        if f.endswith('.html'):
+            result.append('https://content.clickhouse.tech/' + f.replace('index.html', ''))
```

Comment:
btw it looks like this domain can be taken from the `BASE_DOMAIN` env variable
it is overridden in non-production environments, but they usually don't have heavy caching, so not so important